eng/native/gen-buildsys.sh

```diff
@@ -92,11 +92,14 @@ if [[ "$build_arch" == "wasm" ]]; then
 fi
 
 # Include CMAKE_USER_MAKE_RULES_OVERRIDE as uninitialized since it will hold its value in the CMake cache otherwise can cause issues when branch switching
-$cmake_command \
-  -G "$generator" \
-  "-DCMAKE_BUILD_TYPE=$buildtype" \
+if ! $cmake_command                       \
+  -G "$generator"                         \
+  "-DCMAKE_BUILD_TYPE=$buildtype"         \
   "-DCMAKE_INSTALL_PREFIX=$__CMakeBinDir" \
-  $cmake_extra_defines \
-  $__UnprocessedCMakeArgs \
-  -S "$1" \
-  -B "$3"
+  $cmake_extra_defines                    \
+  $__UnprocessedCMakeArgs                 \
+  -S "$1"                                 \
+  -B "$3"; then
+    echo "CMake configuration failed possibly due to older version of CMake in use. Please install v3.14.5 or newer from https://www.cmake.org/download/."
```

Comment:
Wouldn't it display this message for any cmake failure (e.g. when someone makes a mistake in a cmake script?)

Reply:
@am11 I wonder if it would be better to check if cmake supports -B option using `cmake --help | grep " \-B "` or something similar in the original location where we were checking for the version instead.

Reply:
That would be better, I agree. I have opened a PR with this idea: https://github.com/dotnet/runtime/pull/37447.